lang/src/org/partiql/lang/planner/GlobalBindings.kt

```diff
@@ -0,0 +1,50 @@
+package org.partiql.lang.planner
+
+import org.partiql.lang.eval.BindingCase
+import org.partiql.lang.eval.BindingName
+
+/** Indicates the result of an attempt to resolve a global binding. */
+sealed class ResolutionResult {
+    /**
+     * A success case, indicates the [uniqueId] of the match to the [BindingName] in the global scope.
+     * Typically, this is defined by the storage layer.
+     */
+    data class GlobalVariable(val uniqueId: String) : ResolutionResult()
+
+    /**
+     * A success case, indicates the [index] of the only possible match to the [BindingName] in a local lexical scope.
+     * This is `internal` because [index] is an implementation detail that shouldn't be accessible outside of this
+     * library.
+     */
+    internal data class LocalVariable(val index: Int) : ResolutionResult()
```

Comment:
Considering that `GlobalBindings.resolve()` is meant to return `GlobalVariable` or `undefined`, should this be here?

Reply:
It's `internal`, so it's not exposed to customers.  Customer code is never intended to resolve local variables, so this fits the situation.  Having this here does simplify a few things in relation in the `LogicalToLogicalResolvedVisitorTransform` class.  Namely, I found this to be preferable to having two variations of this `sealed` class: one that is public and only has `GlobalVariable` and `Undefined`, and another that is private and has `GlobalVariable`, `Undefined` as well as  `LocalVariable`. 

Making one member of a `public` sealed class `internal` is perhaps a little unusual, but unusual doesn't equate to undesirable, necessarily.  Was there a specific concern you had with this approach?

Reply:
I wasn't worried about exposing it, I just thought this data class doesn't belong here. I think your explanation makes sense, so let's leave it as is.